daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -1260,6 +1261,33 @@ private[lf] object Speedy {
     }
   }
 
+  /** KCloseExercise. Marks an open-exercise which needs to be closed. Either:
+    * (1) by 'endExercise' if this continuation is entered normally, or
+    * (2) by 'abortExercise' if we unwind the stack through this continuation (TODO 8020)
+    */
+  private[speedy] final case class KCloseExercise(
+      machine: Machine,
+      end_templateId: TypeConName,
+  ) extends Kont
+      with SomeArrayEquals {
```

Comment:
```suggestion
  ) extends Kont {
```

`SomeArrayEquals` if for case classs that contain `Array`.

Reply:
thanks; removed.